canister/src/types.rs

```diff
@@ -128,12 +140,15 @@ impl Block {
     // Computes the difficulty given a block's target.
     // The definition here corresponds to what is referred as "bdiff" in
     // https://en.bitcoin.it/wiki/Difficulty
-    //
-    // NOTE: This is dead code temporarily and will be used in an upcoming PR.
-    #[allow(dead_code)]
     fn target_difficulty(network: Network, target: Uint256) -> u64 {
         (ic_btc_validation::max_target(&network.into()) / target).low_u64()
     }
+
+    #[cfg(test)]
+    pub fn with_mock_difficulty(mut self, mock_difficulty: u64) -> Self {
+        self.mock_difficulty = Some(mock_difficulty);
+        self
+    }
```

Comment:
nit: I think this method belongs to the `BlockBuilder` in `test_utils.rs`. I'd make the `mock_difficulty` field public, and move this method into the `BlockBuilder`.

Reply:
I guess it is unnecessary, and we will need to change a lot of code since we are not always using this method after BlockBuilder.

Reply:
I don't have a strong opinion, but does it really change a lot of code? This method is called in ~10 places, and if we move it to the BlockBuilder, the updates we need to make in these places are trivial and would only take a few minutes right? If yes, then I think it's worth it to keep our block builder methods all in one place.